### PR TITLE
[CARBONDATA-743] Remove redundant CarbonFilters file

### PR DESCRIPTION
**Pull request by lionelcao**
Remove redundant CarbonFilters file in Spark2 and keep the one in spark common.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Can one of the admins verify this patch?

**Comment by chenliang613** [comment]
add to whitelist

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1014/

File: integration/spark2/src/main/scala/org/apache/carbondata/spark/CarbonFilters.scala [hunk 1]
```diff
@@ -1,397 +0,0 @@
...
-  }
-
-  def processExpression(exprs: Seq[Expression],
-      attributesNeedToDecode: java.util.HashSet[AttributeReference],
-      unprocessedExprs: ArrayBuffer[Expression],
-      carbonTable: CarbonTable): Option[CarbonExpression] = {
-    def transformExpression(expr: Expression, or: Boolean = false): Option[CarbonExpression] = {
-      expr match {
-        case or@ Or(left, right) =>
-          val leftFilter = transformExpression(left, or = true)
```
Comment (chenliang613) [inline integration/spark2/src/main/scala/org/apache/carbondata/spark/CarbonFilters.scala]: The below code is different with spark-common's CarbonFilter.scala, please consider if need to keep consistent.
```
val leftFilter = transformExpression(left, or = true)
          val rightFilter = transformExpression(right, or = true)
```

File: integration/spark2/src/main/scala/org/apache/carbondata/spark/CarbonFilters.scala [hunk 2]
```diff
@@ -1,397 +0,0 @@
...
-          if (!or) {
-            others.collect {
-              case attr: AttributeReference =>
-                attrList.add(AttributeReferenceWrapper(aliasMap.getOrElse(attr, attr)))
-            }
-          }
-          None
-      }
-    }
-    filters.flatMap(translate(_)).toArray
```
Comment (chenliang613) [inline integration/spark2/src/main/scala/org/apache/carbondata/spark/CarbonFilters.scala]: The below code is different with spark-common's CarbonFilter.scala, please consider if need to keep consistent.
`filters.flatMap(translate(_, false)).toArray`

File: integration/spark2/src/main/scala/org/apache/carbondata/spark/CarbonFilters.scala [hunk 3]
```diff
@@ -1,397 +0,0 @@
...
-          if (!or) {
-            others.collect {
-              case attr: AttributeReference => attributesNeedToDecode.add(attr)
-            }
-            unprocessedExprs += others
-          }
-          None
-      }
-    }
-    exprs.flatMap(transformExpression(_)).reduceOption(new AndExpression(_, _))
```
Comment (chenliang613) [inline integration/spark2/src/main/scala/org/apache/carbondata/spark/CarbonFilters.scala]: The below code is different with spark-common's CarbonFilter.scala, please consider if need to keep consistent.
`exprs.flatMap(transformExpression(_)).reduceOption(new AndExpression(_, _))`

Comment (lionelcao) [inline integration/spark2/src/main/scala/org/apache/carbondata/spark/CarbonFilters.scala]: The 2nd parameter in transformExpression() is false by default thus transformExpression(_) equals to transformExpression(_, false).

File: integration/spark2/src/main/scala/org/apache/carbondata/spark/CarbonFilters.scala (hunk 2 above)
Comment (lionelcao) [inline integration/spark2/src/main/scala/org/apache/carbondata/spark/CarbonFilters.scala]: The 2nd parameter in translate() is false by default thus translate(_) equals to translate(_, false).

File: integration/spark2/src/main/scala/org/apache/carbondata/spark/CarbonFilters.scala (hunk 1 above)
Comment (lionelcao) [inline integration/spark2/src/main/scala/org/apache/carbondata/spark/CarbonFilters.scala]: transformExpression(left, or = true)  equals to transformExpression(left, true) here.

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1020/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1021/

**Pull request closed without merging**